signal_timeout_block.py

```diff
@@ -38,18 +36,26 @@ def __init__(self):
         self._jobs = defaultdict(dict)
         self._jobs_locks = defaultdict(Lock)
         self._repeatable_jobs = defaultdict(dict)
+        self._single_jobs = defaultdict(dict)
         self._persistence_scheduled = Event()
 
     def persisted_values(self):
         """Use persistence mixin"""
-        return ["_repeatable_jobs"]
+        return ["_repeatable_jobs", "_single_jobs"]
 
     def start(self):
         super().start()
         # Schedule persisted jobs
         for key, intervals in self._repeatable_jobs.items():
             for interval, job in intervals.items():
+                if not isinstance(job, Signal):  # handle bug in safepickle
+                    job = Signal(job)
```

Comment:
This is potentially dangerous, if `job` is something else like an `int` for example this will fail. We honestly should probably wrap up all of this persistence loading stuff in a method and call it inside of a try/except. If we hit an exception, log a warning saying we may not have loaded everything from persistence, include the exception info.

Reply:
I thought about that. In the case that persistence loads something other than `Signal` or `dict`, should it fail while trying to load it, or when it tries to run that job? In the example log you posted, we didn't know that we had loaded an invalid object until `interval` had elapsed and the job ran.

As it is here, the block will fail to start. Catching that is probably better, but I think this is the right place for the check.